profiler/proftest/proftest.go

```diff
@@ -466,17 +466,19 @@ func parseBackoffDuration(line string) (time.Duration, error) {
 	return backoff, nil
 }
 
-// PollForAndReturnSerialOutput polls serial port 2 of the GCE instance specified by
-// inst and returns when the finishString appears in the serial output
-// of the instance, or when the context times out.
-func (tr *GCETestRunner) PollForAndReturnSerialOutput(ctx context.Context, inst *InstanceConfig, finishString, errorString string) (string, error) {
+// PollAndLogSerialPort polls serial port 2 of the given GCE instance and
+// returns when the finishString appears in the serial output of the instance,
+// or when the context times out. It logs the serial output of the instance
+// using the specified log function and returns the serial output in the first
+// return value.
+func (tr *GCETestRunner) PollAndLogSerialPort(ctx context.Context, inst *InstanceConfig, finishString, errorString string, logf func(string, ...interface{})) (string, error) {
```

Comment:
Instead of adding another method you may consider adding logf as a field on the sturct that can be optionally set.

Reply:
I didn't like a new parameter either and thought about using InstanceConfig too, but this parameter is only needed for this method so it seemed better to have it just here.